libc/string_errsig.c

```diff
@@ -0,0 +1,228 @@
+{
+	for (int i = 0; i < error_codes_len; i++) {
+		errno = 0;
+		TEST_ASSERT_NOT_NULL(strerror(error_codes[i]));
+
+		if (i != 0) {
+			TEST_ASSERT_NOT_NULL(strerror(error_codes[i - 1]));
+			TEST_ASSERT_NOT_EQUAL_INT(0, strcmp(strerror(error_codes[i - 1]), strerror(error_codes[i])));
+		}
+
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(string_errsign, strerror_zero)
+{
+	errno = 0;
+	TEST_ASSERT_NOT_NULL(strerror(0));
+	TEST_ASSERT_EQUAL_INT(0, errno);
+}
+
+
+TEST(string_errsign, strerror_invaild)
+{
+	const int values[] = { INT_MIN, INT_MIN / 2, INT_MIN / 4, -1024, -256, 256, 1024, INT_MAX / 4, INT_MAX / 2, INT_MAX };
+	for (int i = 0; i < (sizeof(values) / sizeof(values[0])); i++) {
+		TEST_ASSERT_NOT_NULL(strerror(values[i]));
+		TEST_ASSERT_NOT_NULL(strstr(strerror(values[i]), "Unknown error"));
+	}
+}
+
+
+TEST(string_errsign, strerror_r_basic)
+{
+/* Disabled because of #694 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/694 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#694 issue");
+#else
+	for (int i = 0; i < error_codes_len; i++) {
+		char prev_buf[MAX_LEN_STRING] = { 0 };
+		char buf[MAX_LEN_STRING] = { 0 };
+
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(0, strerror_r(error_codes[i], buf, sizeof(buf)));
+
+		if (i != 0) {
+			TEST_ASSERT_EQUAL_INT(0, strerror_r(error_codes[i - 1], prev_buf, sizeof(prev_buf)));
+			TEST_ASSERT_NOT_EQUAL_INT(0, strcmp(buf, prev_buf));
+		}
+
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+#endif
+}
+
+
+TEST(string_errsign, strerror_r_zero)
+{
+/* Disabled because of #694 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/694 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#694 issue");
+#else
+	char buf[MAX_LEN_STRING];
+	errno = 0;
+	TEST_ASSERT_EQUAL_INT(0, strerror_r(0, buf, sizeof(buf)));
+	TEST_ASSERT_EQUAL_INT(0, errno);
+#endif
+}
+
+
+TEST(string_errsign, strerror_r_invaild)
+{
+/* Disabled because of #694 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/694 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#694 issue");
+#else
+	const int values[] = { INT_MIN, INT_MIN / 2, INT_MIN / 4, -1024, -256, 256, 1024, INT_MAX / 4, INT_MAX / 2, INT_MAX };
+	for (int i = 0; i < (sizeof(values) / sizeof(values[0])); i++) {
+		char buf[MAX_LEN_STRING] = { 0 };
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(EINVAL, strerror_r(values[i], buf, sizeof(buf)));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+#endif
+}
+
+
+TEST(string_errsign, strerror_r_overflow)
+{
+/* Disabled because of #694 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/694 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#694 issue");
+#else
+	char buf[1];
+	char buf2[MAX_LEN_STRING];
+	for (int i = 0; i < error_codes_len; i++) {
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(ERANGE, strerror_r(error_codes[i], buf, sizeof(buf)));
+		TEST_ASSERT_EQUAL_INT(ERANGE, strerror_r(error_codes[i], buf2, 2));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+#endif
+}
+
+
+TEST(string_errsign, strsignal_basic)
```

Comment:
IMHO the strsignal tests should be moved to the other file (and the array with signals above removed)

Reply:
We would rather name files using specific header prefixes. `strsignal` is from string.h, so it's in string_errsig.c, `psignal` from signal.h so it's tested in signal.c, (or some other file with signal prefix)

Reply:
If you feel the need to, than do so - but please keep only one copy of `signal_codes` array (and other arrays of this type in the future) as any changes to signal definitions would need to be introduced in multiple places now (and this obviously saves some `.rodata` size ;))